modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/runtime/source/ExtensionMessageSourceTestCase.java

```diff
@@ -169,6 +186,64 @@ public void before() throws Exception
     }
 
     @Test
+    public void getMetadataUsesClassLoaderModelProperty() throws MetadataResolvingException
+    {
+        final MetadataResult result = mock(MetadataResult.class);
```

Comment:
this is testing the implementation not the behaviour. If what you want is to make sure that the right classloader is used, then you can just use a custom Answer object that verifies the TCCL